Source/MQTTnet/MqttApplicationMessage.cs

```diff
@@ -48,11 +49,21 @@ public sealed class MqttApplicationMessage
         public uint MessageExpiryInterval { get; set; }
 
         /// <summary>
-        ///     Gets or sets the payload.
-        ///     The payload is the data bytes sent via the MQTT protocol.
+        /// Get or set Payload array
         /// </summary>
         public byte[] Payload { get; set; }
 
+        /// <summary>
+        /// Get or set the offset of Payload
+        /// </summary>
+        public int PayloadOffset { get; set; }
+
+        /// <summary>
+        /// Get or set the effective number of bytes of Payload
+        /// Leaving null means equal to the total length of Payload minus PayloadOffset
+        /// </summary>
+        public int? PayloadCount { get; set; } 
```

Comment:
When new members are added to this type, then there could also be
```c#
public ReadOnlySpan<byte> Span => Payload.AsSpan(PayloadOffset, PayloadLength);
```
as convenience property to get the span of the payload.

Or is a ROM more suitable? Or both?
Then it's too much and just leave it out?